data/scripts/creaturescripts/bestiary_kills.lua

```diff
@@ -0,0 +1,47 @@
+local PARTY_MEMBERS_ENABLED = false
+
+local function kill(player, monsterName, raceId, unlockLimits)
+    local oldKills = player:getBestiaryKills(raceId)
+    player:addBestiaryKills(raceId)
+    local newKills = player:getBestiaryKills(raceId)
+    for _, limit in pairs(unlockLimits) do
+        if oldKills < limit and newKills >= limit then
+            player:sendTextMessage(MESSAGE_EVENT_DEFAULT, string.format("You unlocked details for the creature %s.", monsterName))
+            player:sendBestiaryMilestoneReached(raceId)
+            break
+        end
+    end
+end
+
+local creatureEvent = CreatureEvent("BestiaryKills")
+
+function creatureEvent.onKill(player, target)
+    local monster = target:getMonster()
+    if not monster then
+        return true
+    end
+
+    local info = monster:getType():getBestiaryInfo()
+    if info.raceId == 0 then
+        return true
+    end
+
+    local monsterName = monster:getName()
+    local unlockLimits = {info.prowess, info.expertise, info.mastery}
+    if PARTY_MEMBERS_ENABLED then
```

Comment:
Note that the system in Tibia is not like that, only party members that actually damaged the creature will get kills counted. If you just stand there watching, it will not count.

To add further complications to the mix, this is only when shared exp is enabled. If it is not enabled, only the highest damaging player gets the kill, just like it would be outside a party.

There's the scenario where both happen (some members of a party damage the creature, some members out of the party damage it).

Reply:
I think it's perfect, it's just what I wanted to read